tests/test_unit/test_roi/test_angles.py

```diff
@@ -0,0 +1,394 @@
+import re
+from collections.abc import Iterable
+from typing import Any
+
+import numpy as np
+import pytest
+import xarray as xr
+
+from movement.kinematics import compute_forward_vector
+from movement.roi import LineOfInterest
+from movement.roi.base import BaseRegionOfInterest
+
+
+def sample_position_array() -> xr.DataArray:
+    """Return a simulated position array to test the egocentric angle.
+
+    The data has time, space, and keypoints dimensions.
+
+    The keypoints are left, right, midpt (midpoint), and wild.
+    The midpt is the mean of the left and right keypoints; the wild keypoint
+    may be anywhere in the plane (it is used to test that the function respects
+    the origin of the forward vector that the user provides).
```

Comment:
```suggestion
    The keypoints are left, right, midpt (midpoint), and wild.
    The midpt is the mean of the left and right keypoints; the wild keypoint
    is used to test that the function respects
    the origin of the forward vector that the user provides.
```
I am confused about the defining aspect of "wild", since also the "left" and "right" keypoints are anywhere in the plane right? I think I would remove that.

But I may be missing something, feel free to ignore.

Reply:
It's to distinguish it's purpose from "midpt". "midpt" is in the middle of the "left" and "right" keypoints, so stands-in for a sensible user case where the `direction` is computed from the forward vector (using "left" and "right") and then the `position` argument (origin of the forward vector) is set to the "midpt".

Of course, there is actually no requirement that the `direction` and `position` argument are related in a meaningful way (though in 99% of use cases we expect them to be!) hence the "wild" keypoint is there to check the method behaves even when given physically-odd information.

Reply:
Have changed it to

```
The midpt is the mean of the left and right keypoints.
The wild keypoint is a point in the plane distinct from the midpt, and is
used to test that the function respects the origin of the forward vector
that the user provides (even if it is physically non-nonsensical).
```